### PR TITLE
[WIP] Support enums

### PR DESCRIPTION
**Pull request by Weakky**
- Implements a proper TS/Flow parser to be able to make valuable comparisons between GraphQL and Model types.

- Support Enum as scalar values (GraphQL enums are always inlined in graphqlgen.ts)

- Support smart code generation/scaffolding:

Given the following GraphQL types:

```graphql
type User {
  id: String!
  penColor: Color!
}

enum Color {
  RED
  GREEN
  BLUE
}
```

and the following model type:

```ts
type User {
  id: string
  penColor: ModelColor
}
```

the resolver for the `penColor` field will be generated as a `defaultResolver` if:

- `ModelColor` is a subset or has the same values as `Color`. That is to say if `ModelColor` is of type:
  - `type ModelColor = 'RED'`
  - `type ModelColor = 'BLUE'`
  - `type ModelColor = 'GREEN'`
  - `type ModelColor = 'RED' | 'BLUE'`
  - `type ModelColor = 'RED' | 'GREEN'`
  - `type ModelColor = 'BLUE' | 'GREEN'`
  - `type ModelColor = 'RED' | 'BLUE' | 'GREEN'` etc..

or if `penColor` is of type:

- `penColor: 'RED'`
- `penColor: 'BLUE'`
- `penColor: 'GREEN'`
- `penColor: 'RED' | 'BLUE'`
- `penColor: 'RED' | 'GREEN'`
- `penColor: 'BLUE' | 'GREEN'`
- `penColor: 'RED' | 'BLUE' | 'GREEN'` etc..

### CODE REVIEW

File: packages/graphqlgen-templates/typescript-yoga/src/generated/graphqlgen.ts [hunk 1]
```diff
@@ -141,8 +142,7 @@ export namespace PostResolvers {
 export namespace UserResolvers {
   export const defaultResolvers = {
     id: (parent: User) => parent.id,
-    name: (parent: User) =>
-      parent.name === undefined || parent.name === null ? null : parent.name,
+    name: (parent: User) => (parent.name === undefined ? null : parent.name),
```
Comment (schickling) [inline packages/graphqlgen-templates/typescript-yoga/src/generated/graphqlgen.ts]: Any reason why `()` is needed here?

File: packages/graphqlgen-templates/typescript-yoga/src/generated/graphqlgen.ts [hunk 2]
```diff
@@ -1,10 +1,11 @@
 // Code generated by github.com/prisma/graphqlgen, DO NOT EDIT.
 
 import { GraphQLResolveInfo } from 'graphql'
-import { Context } from '../types'
 import { Post } from '../types'
 import { User } from '../types'
 
+import { Context } from '../types'
```
Comment (schickling) [inline packages/graphqlgen-templates/typescript-yoga/src/generated/graphqlgen.ts]: Let's remove the newline here.

File: packages/graphqlgen/src/generators/common.ts [hunk 3]
```diff
@@ -90,37 +129,60 @@ export function getModelName(type: GraphQLType, modelMap: ModelMap): string {
...
+  graphQLField: GraphQLTypeField | undefined,
+  modelField: FieldDefinition | undefined,
+  args: GenerateArgs,
+) {
+  if (graphQLField === undefined) {
+    return false
+  }
 
-  if (!graphQLField) {
+  if (modelField == undefined) {
```
Comment (schickling) [inline packages/graphqlgen/src/generators/common.ts]:
```suggestion
  if (modelField === undefined) {
```

**Comment by schickling** [comment]
I've released a first version which can be installed via `graphqlgen@beta`. Please report if you run into any problems 🙏

File: packages/graphqlgen-templates/typescript-yoga/src/generated/graphqlgen.ts (hunk 1 above)
Comment (Weakky) [reply packages/graphqlgen-templates/typescript-yoga/src/generated/graphqlgen.ts]: The parenthesis aren't in the code generator. This must be done by prettier somehow 😕

File: packages/graphqlgen/src/generators/common.ts (hunk 3 above)
Comment (Weakky) [reply packages/graphqlgen/src/generators/common.ts]: Good catch!

**Pull request merged**